### ISSUE TITLE
class Blockchain.find_forks

### ISSUE THREAD

**Issue by marius-j-i** [ted92/bitcoin-mining-competition#1]
The `find_forks` method of the Blockchain class returns immediately after the 1st iteration of the for loop. 
It seems like a one-too-many indentation levels. 
The method does not appear to be referenced in the code but should be fixed just in case.

```py
def find_forks(self):
    """
    find all the forked branches in the blockchain
    :return: dictionary {'hash' : <number of blocks pointing at it>}
    """
    prev_hash_dict = dict()
    for k in self.chain:
        p_hash = self.chain[k].prev
        if p_hash in prev_hash_dict:
            prev_hash_dict[p_hash] += 1
        else:
            prev_hash_dict[p_hash] = 1
        return prev_hash_dict
```

**Comment by ted92** [comment]
Ups. That must have been an oversight, and the reason why I was trying so badly the last day to find out why some forks were not added to the non-main chain. Thank you for that. I will look into it more tomorrow morning.Enrico TedeschiSent from my iPhoneOn 15 Mar 2023, at 21:36, Marius Ingebrigtsen ***@***.***> wrote:﻿
The find_forks method of the Blockchain class returns immediately after the 1st iteration of the for loop.
It seems like a one-too-many indentation levels.
The method does not appear to be referenced in the code but should be fixed just in case.
def find_forks(self):
    """
    find all the forked branches in the blockchain
    :return: dictionary {'hash' : <number of blocks pointing at it>}
    """
    prev_hash_dict = dict()
    for k in self.chain:
        p_hash = self.chain[k].prev
        if p_hash in prev_hash_dict:
            prev_hash_dict[p_hash] += 1
        else:
            prev_hash_dict[p_hash] = 1
        return prev_hash_dict

**Comment by ted92** [comment]
You are right, that was an oversight, however that method is not used anymore, it needs to be deleted. I replied to you earlier via mail but it is better from the website. This method was used to find forked blocks in the chain for later confirmation, now deprecated and replaced with 
```python
confirm_chain()
``` 
in the Blockchain class.

Please let me know if you spot any other issues.
